### ISSUE TITLE
General purpose tab

### ISSUE THREAD

**Issue by oskretc** [fman-community/fman#385]
A bit related to #34 

A general purpose tab that can be somehow pinned to different places. (Like F12 in chrome)

Basically, an empty canvas for plugins to use. 

Use cases:

- Search results
- Text Preview
- History
- Favorites
- File Details
- etc, etc, etc.

**Comment by mherrmann** [comment]
Thank you for the suggestion. The following use cases you mentioned can already be implemented via [fman's API for custom file systems](https://fman.io/blog/fmans-new-file-system-api/):

 * Search results
 * History
 * Favorites

That leaves:

 * Text Preview
 * File Details
 * etc.

The "empty canvas" you describe entails that plugins should be able to draw their own widgets (buttons, lists, text, graphics? etc.). fman's [current stance](https://fman.io/docs/plugins-caveats) towards this is that this will not be supported (search for "Qt" on the page). The main reasons for this are: I would have to guarantee support of a hugely complex API for drawing widgets until the end of time. I would never be able to (say) upgrade to a different GUI framework (eg. a theoretical next version of Qt). Second, I had zero chance of guaranteeing that the plugins are well-behaved. That is, that they don't break things in fman, such as the UI, or make it unresponsive etc. Finally, it would take away from fman's minimalism, something I believe is one of its defining features. I agree at the moment fman is _too_ minimalistic. But the Text Preview you mention in particular is something I'm very much against. It's a complex feature to implement. It will be impossible to make as good as dedicated Preview apps that already exist on the market. Also, where does it stop? Text preview first. But then images, video, ... sound? I think fman should integrate with _existing_ solutions, not reinvent the wheel. The reasoning is the same as for a terminal embedded in fman (#39).

So I'll leave this issue open and encourage you to look at the API for custom file systems :-) And I'm open to reconsider my stance if this feature receives a lot of votes. But until then, I believe there are more valuable fish to fry.

**Comment by oskretc** [comment]
I totally agree with the minimalistic approach, that is one of the features I like it so much. 

Having said that, I do think an auxiliary tab will be much appreciated it. Perhaps "empty canvas" was a strong word and maybe it scare you :stuck_out_tongue: , but seriusly a nice compromise will be to limit the canvas to only text with links support so a user could implement its own favorites, launcher, functions etc.. 

Anyway, lets see what the community has to say about it.

I did found a plugin that implements a preview (text and images) 
https://github.com/joshas/Viewer

**Comment by mherrmann** [comment]
^^ yes, I was scared ;) Text with links sounds better.

@joshas has had to endure many warnings "this is not officially supported and many break in the future" from me ;) I still think it's cool he did it; I just needed to say this as the "responsible" maintainer of the project.

**Comment by o314** [comment]
I use vivaldi web panels for devdocs, history, bookmark. 
That's quite useful.
